bin/cheritest/cheritest_vm.c

```diff
@@ -337,31 +339,48 @@ cheritest_vm_cap_share_sigaction(const struct cheri_test *ctp __unused)
 	if (pid == -1)
 		cheritest_failure_errx("Fork failed; errno=%d", errno);
 
+	/*
+	 * Note: we call __sys_sigaction directly here, since the libthr
+	 * _thr_sigaction has a shadow list for the sigaction values
+	 * (per-process) and therefore does not read the new value installed by
+	 * the child process forked with RFSIGSHARE.
+	 */
 	if (pid == 0) {
-		void * __capability passme;
+		void *__capability passme;
 		struct sigaction sa;
 
 		bzero(&sa, sizeof(sa));
 
 		/* This is a little abusive, but shows the point, I think */
 
 		passme = CHERITEST_CHECK_SYSCALL(mmap(0, PAGE_SIZE,
-				PROT_READ | PROT_WRITE | PROT_EXEC, MAP_ANON, -1, 0));
+		    PROT_READ | PROT_WRITE | PROT_EXEC, MAP_ANON, -1, 0));
 		sa.sa_handler = passme;
-		sa.sa_flags = 0;
 
-		CHERITEST_CHECK_SYSCALL(sigaction(SIGUSR1, &sa, NULL));
+		CHERITEST_CHECK_SYSCALL(__sys_sigaction(SIGUSR1, &sa, NULL));
+
+		/* Read it again and check that we get the same value back. */
+		CHERITEST_CHECK_SYSCALL(__sys_sigaction(SIGUSR1, NULL, &sa));
+		fprintf(stderr, "child value read from sigaction(): ");
+		CHERI_FPRINT_PTR(stderr, sa.sa_handler);
+		CHERITEST_CHECK_EQ_CAP(sa.sa_handler, passme)
```

Comment:
```suggestion
		CHERITEST_CHECK_EQ_CAP(sa.sa_handler, passme);
```

Reply:
Hum how did this compile for me? Well fix